scripts/patterns-reorg/models.py

```diff
@@ -0,0 +1,115 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may obtain a
+# copy of this license in the LICENSE file in the root directory of this source
+# tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this copyright
+# notice, and modified files need to carry a notice indicating that they have
+# been altered from the originals.
+
+from __future__ import annotations
+
+import dataclasses
+from dataclasses import dataclass, field
+from pathlib import PurePath, Path
+
+
+@dataclass(frozen=True)
+class Entry:
+    title: str
+    slug: str | None = field(default=None, kw_only=True)
+    children: tuple[Entry, ...] = field(default_factory=lambda: (), kw_only=True)
```

Comment:
FYI `tuple[X, ...]` is the immutable version of `list[X]`. We need it to be immutable to set `frozen=True`. I originally was using `list` and didn't care about immutability, but it resulted in a bug with `filter_entries`, so I switched to this style.